R/augsynth.R

```diff
@@ -52,22 +52,30 @@ single_augsynth <- function(form, unit, time, t_int, data,
     synth_data <- do.call(format_synth, wide)
 
     
-
+    treated_unit <- data %>% filter(!!trt == 1) %>% distinct(!!unit) %>% pull(!!unit)
+    control_units <- data %>% filter(!!unit != treated_unit) %>% distinct(!!unit) %>% pull(!!unit)
+    
+    
     ## add covariates
     if(length(form)[2] == 2) {
         Z <- extract_covariates(form, unit, time, t_int, data, cov_agg)
     } else {
         Z <- NULL
     }
-
+    
     # fit augmented SCM
     augsynth <- fit_augsynth_internal(wide, synth_data, Z, progfunc, 
                                       scm, fixedeff, ...)
-
+    
     # add some extra data
     augsynth$data$time <- data %>% distinct(!!time) %>% pull(!!time)
     augsynth$call <- call_name
     augsynth$t_int <- t_int 
+    
+    if("weights" %in% names(augsynth)) {
```

Comment:
Is there an instance where `augsynth` wouldn't have a `weights` variable?

Reply:
Nope, this is a relic of some debugging I did that I kept. The if statement has been removed.